OpenRA.Mods.Common/Traits/Player/PlaceBeacon.cs

```diff
@@ -41,6 +41,9 @@ public class PlaceBeaconInfo : TraitInfo
 		[SequenceReference(nameof(BeaconImage))]
 		public readonly string CircleSequence = "circles";
 
+		[CursorReference]
+		public readonly string BeaconCursor = "ability";
```

Comment:
This has nothing to do with this trait. I suggest you put this on `BeaconOrderButtonLogic` instead, which can also read in YAML defined fields.

Reply:
I don't believe that `[CursorReference]` works on widgetlogic, only with `TraitInfo`. This trait holds data related to placing beacons, so the cursor to use when placing a beacon has quite a lot to do with it!

Reply:
There are two other global order buttons similar to the beacon one: power down and sell button. Both have their cursors hardcoded in `PowerDownOrderGenerator` and `SellOrderGenerator` respectively. **But** `SellOrderGenerator` at least looks at `SellableInfo.Cursor`, so there's already a precedent set:

```
[CursorReference]
[Desc("Cursor to display when the sell order generator hovers over this actor.")]
public readonly string Cursor = "sell";
```

https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Orders/GlobalButtonOrderGenerator.cs#L93-L103There are two other global order buttons similar to the beacon button: power down and sell button. Both have their cursors hardcoded in `PowerDownOrderGenerator` and `SellOrderGenerator` respectively. **But** `SellOrderGenerator` at least looks at `SellableInfo.Cursor`, so there's already a precedent set:

```
[CursorReference]
[Desc("Cursor to display when the sell order generator hovers over this actor.")]
public readonly string Cursor = "sell";
```

https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Orders/GlobalButtonOrderGenerator.cs#L93-L103

Reply:
`SellOrderGenerator` is also using a hard-coded value at the moment. Would be nice to see that unhardcoded as well.